### PR TITLE
Bugfix/extending auth service

### PR DESCRIPTION
**Pull request by nvolodymyr**

### CODE REVIEW

File: src/app/component/core/header-new/header-new.component.html [hunk 1]
```diff
@@ -73,7 +73,7 @@
                         </li>
                         </ng-template>
                         <ng-template #userNotLoggerInSmall>
-                          <li class="mobile-vis name tertiary-global-button">Nadiia Steblivets</li>
+                          <li class="mobile-vis name tertiary-global-button">{{name}}</li>
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.html:76]: Please, add spases between brackets

File: src/app/component/core/header-new/header-new.component.html [hunk 2]
```diff
@@ -137,7 +137,7 @@
             <div id="user-avatar-wrapper">
              <ul [class.add-shadow]="dropdownVisible">
                <li class="tertiary-global-button">
-                 <a (click)="toggleDropdown()">Nadiia Steblivets
+                 <a (click)="toggleDropdown()">{{name}}
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.html:140]: Sane as line 76

File: src/app/component/core/header-new/header-new.component.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ import { LanguageService } from '../../../i18n/language.service';
 import { Language } from '../../../i18n/Language';
 import { SignInNewComponent } from '../../auth/sign-in-new/sign-in-new.component';
 import { NewSignUpComponent } from '../../auth/new-sign-up/new-sign-up.component';
+import {UserOwnAuthService} from '../../../service/auth/user-own-auth.service';
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.ts:18]: Please add spaces between brackets

File: src/app/component/core/header-new/header-new.component.ts [hunk 4]
```diff
@@ -41,16 +42,18 @@ export class HeaderNewComponent implements OnInit {
               private userService: UserService,
               private achievementService: AchievementService,
               private habitStatisticService: HabitStatisticService,
-              private languageService: LanguageService) {
+              private languageService: LanguageService,
+              private  userOwnAuthService: UserOwnAuthService) {
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.ts:46]: Please remove 1 space between access modificator and the variable name

File: src/app/component/eco-news/news-list/news-list.component.html [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <div>
   <div class="flex-container">
     <app-filter-news (gridOutput)="getFilterData($event)"></app-filter-news>
-    <a [routerLink]="'/create-news'">
+    <a [routerLink]="'/create-news'"  *ngIf="isLoggedIn">
```
Comment (RomanDeveloperAcc) [inline src/app/component/eco-news/news-list/news-list.component.html:4]: Please, write attributes in separate rows

File: src/app/service/auth/user-own-auth.service.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import {Observable, Subject} from 'rxjs';
+import {UserSuccessSignIn} from '../../model/user-success-sign-in';
```
Comment (RomanDeveloperAcc) [inline src/app/service/auth/user-own-auth.service.ts:3]: Please add spaces between brackets

File: src/app/service/auth/user-own-sign-in.service.ts [hunk 7]
```diff
@@ -22,10 +22,11 @@ export class UserOwnSignInService {
   }
 
   public saveUserToLocalStorage(data: UserSuccessSignIn) {
-    this.localStorageService.setFirstName(data.firstName);
+    this.localStorageService.setFirstName(data.name);
     this.localStorageService.setAccessToken(data.accessToken);
     this.localStorageService.setRefreshToken(data.refreshToken);
     this.localStorageService.setUserId(Number(data.userId));
+
```
Comment (RomanDeveloperAcc) [inline src/app/service/auth/user-own-sign-in.service.ts:29]: Please remove empty line

File: src/app/component/core/header-new/header-new.component.html (hunk 1 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.html:76]: fixed

File: src/app/component/core/header-new/header-new.component.ts (hunk 3 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.ts:18]: +

File: src/app/component/core/header-new/header-new.component.ts (hunk 4 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.ts:46]: fix

File: src/app/component/eco-news/news-list/news-list.component.html (hunk 5 above)
Comment (nvolodymyr) [reply src/app/component/eco-news/news-list/news-list.component.html:4]: fix

File: src/app/service/auth/user-own-auth.service.ts (hunk 6 above)
Comment (nvolodymyr) [reply src/app/service/auth/user-own-auth.service.ts:3]: fix

File: src/app/service/auth/user-own-sign-in.service.ts (hunk 7 above)
Comment (nvolodymyr) [reply src/app/service/auth/user-own-sign-in.service.ts:29]: fix

File: src/app/service/auth/user-own-auth.service.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserOwnAuthService {
+  public credentialDataSubject = new Subject<any>();
+
```
Comment (RomanDeveloperAcc) [inline src/app/service/auth/user-own-auth.service.ts:9]: Please, return the constructor method

Comment (nvolodymyr) [reply src/app/service/auth/user-own-auth.service.ts:9]: fixed

**Pull request merged**